src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxIndexPersistence.cs

```diff
@@ -140,16 +141,19 @@ private void CreatePersistentDictionary(IndexingStatsAggregator indexingStatsAgg
             queryContext.OpenReadTransaction();
 
             using var tx = indexContext.OpenWriteTransaction();
-            
+
+            // RavenDB-21043: Since 32bits systems are resource starved, we are only allowing to allocate 128Mb of memory
+            // for training indexing dictionaries. As soon as 128 Mb are allocated by the training process, the enumerator stops.
+            var maxAllocatedMemory = _index.DocumentDatabase.Is32Bits ? new Size(128, SizeUnit.Megabytes) : default;
```

Comment:
See my above comment, let's introduce a config option and set the default accordingly depending on the platform instead of checking that here.